processor/k8sattributesprocessor/processor.go

```diff
@@ -107,22 +107,21 @@ func (kp *kubernetesprocessor) processLogs(ctx context.Context, ld plog.Logs) (p
 
 // processResource adds Pod metadata tags to resource based on pod association configuration
 func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pcommon.Resource) {
-	podIdentifierKey, podIdentifierValue := extractPodID(ctx, resource.Attributes(), kp.podAssociations)
-	if podIdentifierKey != "" {
-		resource.Attributes().InsertString(podIdentifierKey, string(podIdentifierValue))
+	podIdentifierValue := extractPodID(ctx, resource.Attributes(), kp.podAssociations)
+	for i := range podIdentifierValue {
+		if podIdentifierValue[i].Source.From == kube.ConnectionSource && podIdentifierValue[i].Value != "" {
+			resource.Attributes().InsertString(k8sIPLabelName, podIdentifierValue[i].Value)
+		}
 	}
-
 	if kp.passthroughMode {
 		return
 	}
 
-	if podIdentifierKey != "" {
-		if pod, ok := kp.kc.GetPod(podIdentifierValue); ok {
-			for key, val := range pod.Attributes {
-				resource.Attributes().InsertString(key, val)
-			}
-			kp.addContainerAttributes(resource.Attributes(), pod)
+	if pod, ok := kp.kc.GetPod(podIdentifierValue); ok {
```

Comment:
I'm a bit concerned about removing the check for zero value here and using empty `PodIdentifier{}` that can be returned from `extractPodID`. Are we sure that `PodIdentifier{}` cannot get to the cache?

Reply:
I reverted this change. TBH, same behavior like empty `PodIdentifier{}` is going to be possible with configuring only one attribute e.g. `namespace` as association source